lib/matplotlib/backends/backend_ps.py

```diff
@@ -1467,6 +1467,7 @@ def convert_psfrags(tmpfile, psfrags, font_preamble, custom_preamble,
 
     command = '%s cd "%s" && dvips -q -R0 -o "%s" "%s" > "%s"'%(precmd, tmpdir,
                 os.path.split(psfile)[-1], os.path.split(dvifile)[-1], outfile)
+    command=command.replace("~","\\string~")
```

Comment:
IMO this is the wrong place (probably because the line number is the same): you changed the command for `dvips`, but your bug occurred in the `latex` [line](https://github.com/Grillard/matplotlib/blob/master/lib/matplotlib/backends/backend_ps.py#L1455) which si now a few lines up.

Otherwise two small additions:
* could you add a small comment why this is needed?
* This should also make sure that only the input file (->`latexfile`) is treated like this, not the complete command line (e.g. the outfile might also contain a `~`, but there it would result in a completely different filename).

Not sure if you need this or not, but just in case: just edit the file again, `git add .`, `git diff` (should show the removal of the wrong fix and includes the new fix), `git commit --amend` (overwrite the last commit) and then `git push -f` (force push to the PR).